### ISSUE TITLE
Seems like fetching data doesn't work at all for me

### ISSUE THREAD

**Issue by Jese44** [warrior25/HA-Nysse#9]
Seems like the nysse schedules do not work anymore and if does randomly show schedules it works for a day max.... I assume the problem is at nysses end? does the integration work on your (or someone elses system?)
in the latest patch what was the modification you did on fetching the data and if i understood correctly it was supposed to work like "trains toilet" compared to the another way you tried for a short while.

![image](https://github.com/warrior25/HA-Nysse/assets/45014337/9294c00c-7998-4b93-a60c-dd50e565b367)
![image](https://github.com/warrior25/HA-Nysse/assets/45014337/1df29045-3b48-4dea-b2b8-1d23ce2a13d2)

**Comment by warrior25** [comment]
Just tested with Home Assistant 2024.5.3 and works fine. It's possible that there's a problem that shows up only after some time passes, I will keep an eye on my instance if anything happens. Please provide logs from the integration if you encounter any errors.

**Comment by Jese44** [comment]
That's odd. I've had problems on shoving schedule ever since the first update that had "new method on fetching data" was implemented (months ago). after that schedules may show on one day of the week. and also one recent problem i had when i set limit of the buses to show as 6, the integration only showed 3 for some reason (and it was day in the middle of the week so every 15 minutes should've been one bus showing)
One thing that could be the problem is my pihole, even though i have never blacklisted any nysse domains and it does sometimes work. 
Do you have any tips on what address(es) should i try to "force whitelist" if that is the case?
I can try to access sites via browser first to check if pihole is the problem, i just dont know what site...

I just enable debug logging for the integration so i will! i restarted the integration -> no errors. restarted HA --> no errors.

Core: 2024.5.1
Supervisor: 2024.05.1
Operating System: 12.2
Frontend: 20240501.0

**Comment by Jese44** [comment]
So since your system works i decided to delete integration (via hacs) and re-install + restart HA (also disabled pi-hole totally from blocking) and integration state this error when trying to add stops: "Failed to fetch stops. Please try again later"

**Comment by warrior25** [comment]
I have managed to replicate this issue. I will look into it further.

**Comment by warrior25** [comment]
Couldn't find any specific issue on my side, which would have caused these kinds of issues. The integration seems to be working correctly as of now. I pushed a new update to make the integration more resilient to errors, which should lessen the cases where the integration needs to be manually reloaded. 

If you're still encountering any issues please have a look at the troubleshooting section of the README or create a new issue.
